### ISSUE TITLE
After restarting 7.5.1: no valid themes are found on this instance (intermittent)

### ISSUE THREAD

**Issue by javieranton-zz** [salesagility/SuiteCRM#6444]
#### Issue
This started happening today after I rebooted my computer
Whenever I try going to SuiteCRM or accessing its REST svc a response of "no valid themes are found on this instance" is returned. This happens intermittently and if I keep reloading it manages to load SuiteCRM some times. But then the issue appears again. I tried rebuilding and repairing but it dind't work

#### Expected Behavior
SuiteCRM should load/ REST svc should reply

#### Actual Behavior
"no valid themes are found on this instance" is displayed in browser. Nothing is logged in logs

#### Steps to Reproduce
No way to reproduce as it started happening after rebooting for no reason

#### Context
It has completely broken SuiteCRM
HIGH

#### Your Environment
* SuiteCRM Version used: 7.5.1
* All browsers & REST svcs
* SQL SERVER & IIS with PHP
* Operating System: Win 7

**Comment by javieranton-zz** [comment]
From what I can tell whatever is failing is happening in include\SugarTheme\SugarTheme.php:buildRegistry()

**Comment by Dillon-Brown** [comment]
Hi @javieranton-zz 

This question may be more suited to the [Community Forums](https://www.suitecrm.com/forum/suite-forum). We use GitHub primarily for the reporting of Core SuiteCRM bugs and leave general questions and setup specific issues to the forums.

Thanks.

**Comment by Dillon-Brown** [comment]
@javieranton-zz Unless you could replicate this on the LTS version (v7.8.22) or 7.10.x as a core issue then it makes more sense to report it on the general forums. This issue will likely be related to your specific setup and environment.

A few possible solutions that i'd recommend trying first:
1. Run a "Quick repair and rebuild" in the admin panel.
2. Do a hard reset of the browser cache.
3. Set permissions (Linux):
 ```
  sudo chown -R www-data:www-data .
  sudo chmod -R 755 .
  sudo chmod -R 775 cache custom modules themes data upload
  sudo chmod 775 config_override.php 2>/dev/null
```

**Comment by javieranton-zz** [comment]
I've tracked down the issue to is_readable($dir) sometimes returning true and some times returning false
It's true that it's likely an environment issue

**Comment by javieranton-zz** [comment]
GOOD NEWS! I was able to fix this

Thanks so much for your help, patience and suggestions. Apologies for being impolite when first posting

I will describe below what fixed it in case anyone encounters this again...

STEPS TO RESOLUTION:
1 - Upgrade PHP to 7.0 (PHP 5 wasn't letting me do step 2)
2 - Very painfully upgrade SuiteCRM (in my case from 7.5.1 to 7.7.9). Very painful process as I got some "no valid theme" msg at a couple points and had to resubmit the post request via browser refresh). I made sure to run this on my backup first just in case

After this the issue is gone

My guess of what may have happened is that my disk is corrupt somehow and new files from upgrade have landed on a safe part of it. Who knows!
